app/models/finder_schema.rb

```diff
@@ -14,74 +20,73 @@ def self.load_from_schema(type)
   end
 
+  attribute :base_path
+  attribute :beta
+  attribute :beta_message
+  attribute :content_id
+  attribute :default_order
+  attribute :description
+  attribute :document_noun
+  attribute :document_title
+  attribute :editing_organisations, default: []
+  attribute :email_filter_by
+  attribute :email_filter_facets, default: []
+  attribute :facets, default: []
+  attribute :filter
+  attribute :format_name
+  attribute :label_text
+  attribute :name
+  attribute :organisations, default: []
+  attribute :open_filter_on_load
+  attribute :parent
+  attribute :phase
+  attribute :related
   attribute :show_summaries, :boolean, default: false
-  attr_writer :editing_organisations, :facets, :taxons
-  attr_reader :document_noun, :related
-  attr_accessor :base_path,
-                :beta,
-                :beta_message,
-                :content_id,
-                :default_order,
-                :description,
-                :document_title,
-                :email_filter_by,
-                :email_filter_facets,
-                :filter,
-                :format_name,
-                :label_text,
-                :name,
-                :open_filter_on_load,
-                :parent,
-                :phase,
-                :signup_content_id,
-                :signup_copy,
-                :signup_link,
-                :subscription_list_title_prefix,
-                :summary,
-                :target_stack,
-                :topics
-
-  def document_noun=(noun)
-    @document_noun = noun
-    if @signup_copy.present?
-      @signup_copy = "You'll get an email each time a #{noun} is updated or a new #{noun} is published."
+  attribute :signup_content_id
+  attribute :signup_copy
+  attribute :signup_link
+  attribute :subscription_list_title_prefix
+  attribute :summary, default: ""
+  attribute :target_stack
+  attribute :taxons, default: []
+  attribute :topics, default: []
+
+  def update(attributes)
+    run_callbacks :update do
+      assign_attributes(attributes)
     end
   end
 
-  def format
-    @filter["format"]
-  end
-
-  def taxons
-    @taxons || []
+  def as_json(options = nil)
+    super.compact.reject { |_k, v| v.blank? }
   end
 
-  def organisations
-    @organisations || []
-  end
-
-  def organisations=(value)
-    @organisations = value.reject(&:empty?)
+  def format
+    filter["format"]
   end
 
-  def editing_organisations
-    @editing_organisations || []
+  def reset_email_alerts
+    assign_attributes(
+      signup_content_id: nil,
+      subscription_list_title_prefix: nil,
+      signup_link: nil,
+      email_filter_by: nil,
+      email_filter_facets: nil,
+    )
   end
 
-  def related=(value)
-    @related = value.nil? ? nil : value.reject(&:empty?)
+  def override_signup_copy
+    if signup_copy.present?
+      _assign_attribute(:signup_copy, "You'll get an email each time a #{document_noun} is updated or a new #{document_noun} is published.")
```

Comment:
If anyone can find a less ugly way of doing this, please holler

Reply:
Worth specifying in the commit message the various things you already tried (`self.<attribute_name> =`, etc). From dev huddle, sounds like you've already tried everything I would have, so no suggestions from me!